sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java

```diff
@@ -785,6 +786,182 @@
+     * {@codesnippet com.azure.ai.formrecognizer.FormRecognizerAsyncClient.beginRecognizeIdDocuments#Flux-long}
+     *
+     * @param idDocument The data of the document to recognize ID Document information from.
+     * @param length The exact length of the data.
+     *
+     * @return A {@link PollerFlux} that polls the recognize ID Document operation until it has completed, has failed,
+     * or has been cancelled. The completed operation returns a list of {@link RecognizedForm}.
+     * @throws FormRecognizerException If recognize operation fails and the {@link AnalyzeOperationResult} returned with
+     * an {@link OperationStatus#FAILED}.
+     * @throws NullPointerException If {@code idDocument} is null.
+     */
+    @ServiceMethod(returns = ReturnType.LONG_RUNNING_OPERATION)
+    public PollerFlux<FormRecognizerOperationResult, List<RecognizedForm>> beginRecognizeIdDocuments(
+        Flux<ByteBuffer> idDocument, long length) {
+        return beginRecognizeIdDocuments(idDocument, length, null);
+    }
+
+    /**
+     * Analyze ID documents using optical character recognition (OCR) and a prebuilt ID trained
+     * model to extract key information from passports and US driver licenses.
+     * See <a href="https://aka.ms/formrecognizer/iddocumentfields">here</a> for fields found on an ID document.
+     * <p>The service does not support cancellation of the long running operation and returns with an
+     * error message indicating absence of cancellation support.</p>
+     *
+     * Note that the {@code idDocument} passed must be replayable if retries are enabled (the default).
+     * In other words, the {@code Flux} must produce the same data each time it is subscribed to.
+     *
+     * <p><strong>Code sample</strong></p>
+     * {@codesnippet com.azure.ai.formrecognizer.FormRecognizerAsyncClient.beginRecognizeIdDocuments#Flux-long-RecognizeIDDocumentOptions}
+     *
+     * @param idDocument The data of the document to recognize ID Document information from.
+     * @param length The exact length of the data.
+     * @param recognizeIDDocumentOptions The additional configurable {@link RecognizeIDDocumentOptions options}
+     * that may be passed when analyzing an ID Document.
+     *
+     * @return A {@link PollerFlux} that polls the recognize ID Document operation until it has completed, has failed,
+     * or has been cancelled. The completed operation returns a list of {@link RecognizedForm}.
+     * @throws FormRecognizerException If recognize operation fails and the {@link AnalyzeOperationResult} returned with
+     * an {@link OperationStatus#FAILED}.
+     * @throws NullPointerException If {@code idDocument} is null.
+     */
+    @ServiceMethod(returns = ReturnType.LONG_RUNNING_OPERATION)
+    public PollerFlux<FormRecognizerOperationResult, List<RecognizedForm>> beginRecognizeIdDocuments(
+        Flux<ByteBuffer> idDocument, long length, RecognizeIDDocumentOptions recognizeIDDocumentOptions) {
```

Comment:
Noticed that the options type is named `RecognizeIDDocumentsOptions` (both letters in `ID` capitalized), while the methods are all named `beginRecognizeIdDocuments` (`Id` with only the first letter capitalized). Should they be made the same for consistency's sake?

Reply:
yes, good find, I will update it to RecognizeIdDocumentsOptions for consistency.